native-sql-engine/cpp/src/codegen/arrow_compute/ext/sort_kernel.cc

```diff
@@ -1399,7 +1399,9 @@ class SortOnekeyKernel : public SortArraysToIndicesKernel::Impl {
     if (nulls_total_ == 0) {
       // if all batches have no null value,
       // we do not need to check whether the value is null
+      ARROW_CHECK_LE(num_batches_, 64 * 1024);
```

Comment:
num_batches_ is controled by partition size eventually. looks limitation of 64K maximal recordbatch is too small.